src/lib/snack-bar/snack-bar-ref.ts

```diff
@@ -84,13 +88,18 @@ export class MatSnackBarRef<T> {
       this._onAction.complete();
     }
 
-    this._afterClosed.next();
-    this._afterClosed.complete();
+    this._afterDismissed.next(this._dismissedWithAction);
```

Comment:
@josephperrott I'm also open to having a `reason` parameter on the `dismiss` method. I didn't go with it now, because it would mean that we'd need some magical string (e.g. `'action'`), because the consumer doesn't usually have access to the action button on the `SimpleSnackBar`.